### PR TITLE
SDIT-2319 Separate search and filter and improve search appearance

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: server/routes/search/searchController.ts [hunk 1]
```diff
@@ -115,10 +118,31 @@ export default class SearchController {
     return res.render('pages/suggestion')
   }
 
-  getPaginationParams(req: Request, page: PageMetaData): LegacyPagination {
-    // const filters = req.session.searchParams.filters ? `&filters=${req.session.searchParams.filters}` : ''
-    // const paginationUrlPrefix = `/search/results?${filters}&`
-    const paginationUrlPrefix = `/search/results?`
+  getSessionFilterString(req: Request): string {
+    const searchParams = new URLSearchParams()
```
Comment (karenmillermoj) [inline server/routes/search/searchController.ts:122]: I am doing a lot of faffing with the filter string - will see if I can tidy this up a bit

File: server/views/macros/filterMacro.njk [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{% macro categoryFilter(params) %}
```
Comment (karenmillermoj) [inline server/views/macros/filterMacro.njk:1]: Just realised not in the filter dir - will move in next pr to sit next to the searchFilterMacro

File: server/routes/search/searchController.ts [hunk 3]
```diff
@@ -115,10 +118,31 @@ export default class SearchController {
...
+      searchParams.append('filters', entry)
+    })
+    return searchParams ? '' : searchParams.toString()
+  }
+
+  getFiltersFromQuery(req: Request): string[] {
+    const { filters } = req.query
+    if (filters) {
+      if (Array.isArray(filters)) {
+        logger.debug('isarray')
```
Comment (petergphillips) [inline server/routes/search/searchController.ts:134]: would be good to remove this debugging at some point

File: server/utils/nunjucksSetup.ts [hunk 4]
```diff
@@ -48,4 +48,57 @@ export default function nunjucksSetup(app: express.Express): void {
   njkEnv.addFilter('acronymsToUpperCase', acronymsToUpperCase)
   njkEnv.addFilter('spaceHyphens', spaceHyphens)
   njkEnv.addFilter('assetMap', (url: string) => assetManifest[url] || url)
+  njkEnv.addFilter('categoryFilters', (searchFilters: string[]) => {
```
Comment (petergphillips) [inline server/utils/nunjucksSetup.ts:51]: would be better to extract this out to a function so that it can be tested

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**